src/modules/fancyzones/lib/FancyZones.cpp

```diff
@@ -238,41 +238,56 @@ void FancyZones::ToggleEditor() noexcept
         m_terminateEditorEvent.reset(CreateEvent(nullptr, true, false, nullptr));
     }
 
-	POINT currentCursorPos;
-	if (!GetCursorPos(&currentCursorPos))
-	{
-		return;
-	}
+    HMONITOR monitor = NULL;
+    UINT dpi_x = 96;
+    UINT dpi_y = 96;
+
+    if (m_settings->GetSettings().use_cursorpos_editor_startupscreen)
+    {
+        POINT currentCursorPos;
+        if (!GetCursorPos(&currentCursorPos))
+        {
+            return;
+        }
+
+        monitor = MonitorFromPoint(currentCursorPos, MONITOR_DEFAULTTOPRIMARY);
+        DPIAware::GetScreenDPIForPoint(currentCursorPos, dpi_x, dpi_y);

```

Comment:
POINT currentCursorPos{};
GetCursorPos(&currentCursorPos);
monitor = MonitorFromPoint(currentCursorPos, MONITOR_DEFAULTTOPRIMARY);
DPIAware::GetScreenDPIForPoint(currentCursorPos, dpi_x, dpi_y);

This will still open the editor on the primary monitor if GetCursorPos fails for whatever reason.

Reply:
sounds reasonable.